packages/tailwindcss/src/variants.ts

```diff
@@ -900,9 +906,16 @@ export function createVariants(theme: Theme): Variants {
 
 function quoteAttributeValue(value: string) {
   if (value.includes('=')) {
+    // Do not allow invalid operators in attribute values
+    if (/[*$^|~]\s+=/.test(value)) {
```

Comment:
I think this condition would break this variant:

```
data-[potato="^_="]
```

Feeling like the better overall solution to this problem should rely on `segment` — do we just need to segment the value on `=` and quote the last chunk if it isn't already quoted? Maybe we don't need to protect people from writing `data-[potato_^_=_foo]` and just let that be user error that generates junk.

Reply:
Reworked it by using `segment` instead of the regex.